datastore/google/cloud/datastore/connection.py

```diff
@@ -220,6 +237,44 @@ def allocate_ids(self, project, request_pb):
                          _datastore_pb2.AllocateIdsResponse)
 
 
+def _grpc_catch_rendezvous(to_call, *args, **kwargs):
```

Comment:
Just a thought I had...
If you have more than one operation that occurs before the result is returned then this function wouldn't work since you can only pass one `to_call`. Plus I think it might clean up the code a little bit.

Also, not a context manger thing specifically, but what would you think about making this a decorator?
`@map_grpc_errors` or something.